frontend/src/pages/ActivityReport/index.js

```diff
@@ -323,6 +323,13 @@ function ActivityReport({
       reportId.current = savedReport.id;
       window.history.replaceState(null, null, `/activity-reports/${savedReport.id}/${currentPage}`);
     } else {
+      // If user has one role set creator role.
+      if (!data.creatorRole && userHasOneRole) {
+        const [onlyUserRole] = user.role;
```

Comment:
This seems to be the best location in code to set the creator role if they only have one.

I'm open to other suggestions.

Reply:
You could remove the if statement and the eslint warning by writing that more declaratively, something like:

```js
const creatorRole = !data.creatorRole && userHasOneRole ? user.role[0] : data.creatorRole;
const updatedFields = findWhatsChanged({ ...data, creatorRole}, formData);
```

Reply:
Agree this way is much cleaner for the linter. C# habits are still hard to change :-).

Will change.